testkit/src/main/scala/org/bitcoins/testkit/core/gen/TLVGen.scala

```diff
@@ -98,14 +111,239 @@ trait TLVGen {
     } yield OracleAnnouncementV0TLV(sig, pubkey, eventTLV)
   }
 
+  def contractInfoV0TLV: Gen[ContractInfoV0TLV] = {
+    for {
+      numOutcomes <- Gen.choose(2, 10)
+      outcomes <- Gen.listOfN(numOutcomes, CryptoGenerators.sha256Digest)
+      totalInput <-
+        Gen
+          .choose(numOutcomes + 1, Long.MaxValue / 10000L)
+          .map(Satoshis.apply)
+    } yield {
+      val satVals = (0 until outcomes.length - 2).toVector.map { _ =>
+        Satoshis(NumberUtil.randomLong(totalInput.toLong))
+      } :+ totalInput :+ Satoshis.zero
+
+      val outcomeValueMap = outcomes.toVector.zip(satVals)
+      ContractInfoV0TLV(outcomeValueMap)
+    }
+  }
+
+  def oracleInfoV0TLV: Gen[OracleInfoV0TLV] = {
+    for {
+      pubKey <- CryptoGenerators.schnorrPublicKey
+      rValue <- CryptoGenerators.schnorrNonce
+    } yield OracleInfoV0TLV(pubKey, rValue)
+  }
+
+  def fundingInputP2WPKHTLV: Gen[FundingInputV0TLV] = {
+    for {
+      prevTx <- TransactionGenerators.realisticTransactionWitnessOut
+      prevTxVout <- Gen.choose(0, prevTx.outputs.length - 1)
+      sequence <- NumberGenerator.uInt32s
+      (spk, _) <- ScriptGenerators.p2wpkhSPKV0
+      newOutput = prevTx.outputs(prevTxVout).copy(scriptPubKey = spk)
+      newPrevTx = prevTx match {
+        case transaction: NonWitnessTransaction =>
+          BaseTransaction(transaction.version,
+                          transaction.inputs,
+                          transaction.outputs.updated(prevTxVout, newOutput),
+                          transaction.lockTime)
+        case wtx: WitnessTransaction =>
+          wtx.copy(outputs = wtx.outputs.updated(prevTxVout, newOutput))
+      }
+    } yield {
+      DLCFundingInputP2WPKHV0(newPrevTx, UInt32(prevTxVout), sequence).toTLV
+    }
+  }
+
+  def fundingInputV0TLV: Gen[FundingInputV0TLV] = {
+    fundingInputP2WPKHTLV // Soon to be Gen.oneOf
```

Comment:
do we not have p2wsh?

Reply:
I mean we could but I didn't want to add anything on this PR that wasn't on adaptor-dlc